src/keys.cc

```diff
@@ -192,6 +231,35 @@ UnitTest test_keys{[]()
     kak_assert(keys == parsed_keys);
     kak_assert(ConstArrayView<Key>{parse_keys("a<c-a-b>c")} ==
                ConstArrayView<Key>{'a', {Key::Modifiers::ControlAlt, 'b'}, 'c'});
+
+    kak_assert(parse_keys("x") == KeyList{ {'x'} });
+    kak_assert(parse_keys("<x>") == KeyList{ {'x'} });
+    kak_assert(parse_keys("<s-x>") == KeyList{ {'X'} });
+    kak_assert(parse_keys("<X>") == KeyList{ {'X'} });
+    kak_assert(parse_keys("X") == KeyList{ {'X'} });
+    kak_assert(parse_keys("<s-up>") == KeyList{ {Key::Modifiers::Shift, Key::Up} });
+    kak_assert(parse_keys("<s-tab>") == KeyList{ {Key::BackTab} });
+
+    try {
+        parse_keys("<s-X>");
+    } catch (runtime_error& err) {
+	kak_assert(err.what() == "Shift modifier only works on special keys and lowercase ASCII");
```

Comment:
Relying on error messages is error prone, wouldn't it be enough to check that an exception is thrown?

Reply:
I think it would make sense to have a custom exception for parsing errors, @mawww ?

Reply:
yeah, introducing a specific exception type, inheriting from runtime_error for that is fine. But frankly, parse_keys has not much other ways to fail than failing to parse a key... Also, introduce a macro for that in unit_test.hh `kak_expect_throw(exception_type, expression)` to keep the unit test easy to follow.